internal/services/servicenetworking/service_networking_application_load_balancer_resource.go

```diff
@@ -0,0 +1,210 @@
+package servicenetworking
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/servicenetworking/2023-05-01-preview/trafficcontrollerinterface"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ApplicationLoadBalancerResource struct{}
+
+type ApplicationLoadBalancerModel struct {
+	Name                   string            `tfschema:"name"`
+	ResourceGroupName      string            `tfschema:"resource_group_name"`
+	Location               string            `tfschema:"location"`
+	ConfigurationEndpoints []string          `tfschema:"configuration_endpoint"`
```

Comment:
I'd like to invite @JackStromberg to provide insights about `what it is used for`. To help determine how to pull this out.

Reply:
@ziyeqf rather than involving additional folks, as the author of the PR, please could you respond to the question which is about the design of the resource and how this translates to user experience. It’s important to get this right rather than introduce breaking changes down the line.

Reply:
okay,after thinking I updated the code